src/domain/post.js

```diff
@@ -16,21 +18,24 @@ export default class Post {
     return new Post(content)
   }
 
-  constructor(content, userId, id, createdAt, postComments) {
+  constructor(content, userId, id, createdAt, postComments, user, profile) {
     this.userId = userId
```

Comment:
you're duplicating data: you're now storing `userId` which is also inside the `user` JS object. You might want to see if you can avoid this, however there might be quite a few client refactoring implications if you remove the userId property now.